src/data/option/basic.lean

```diff
@@ -119,6 +149,21 @@ lemma comp_map (h : β → γ) (g : α → β) (x : option α) :
   option.map g ∘ option.map f = option.map (g ∘ f) :=
 by { ext x, rw comp_map }
 
+lemma bind_map_comm {α β} {x : option (option α) } {f : α → β} :
+  x >>= option.map f = x.map (option.map f) >>= id :=
+by { cases x; simp }
+
+lemma join_map_eq_map_join {f : α → β} {x : option (option α)} :
+  (x.map (option.map f)).join = x.join.map f :=
+by { rcases x with _ | _ | x; simp }
+
+@[simp] lemma join_join {x : option (option (option α))} :
```

Comment:
I don't think this lemma should be marked `simp`

Reply:
In current mathlib,
```
src/data/list/basic.lean:lemma join_join (l : list (list (list α))) : l.join.join = (l.map join).join :=
src/data/seq/seq.lean:@[simp] theorem join_join (SS : seq (seq1 (seq1 α))) :
src/data/seq/wseq.lean:@[simp] theorem join_join (SS : wseq (wseq (wseq α))) :
```
and I modeled the lemma on the majority. Is `option` more like `list`?

Reply:
You can follow the `list` example. I don't think the `seq` and `wseq` versions should be simp either but I expect it probably breaks some proofs.

Reply:
```suggestion
lemma join_join {x : option (option (option α))} :
```